src/main/resources/db/migration/V1_71__appointment_references_referral.sql

```diff
@@ -0,0 +1,28 @@
+alter table appointment
+    add column referral_id UUID,
+    add constraint fk__appointment__referral foreign key (referral_id) references referral;
+
+Update appointment apo
+set referral_id = (
+    select ref.id
+    from referral ref
+             inner join action_plan act on ref.id = act.referral_id
+             inner join action_plan_session aps on aps.action_plan_id = act.id
+             inner join action_plan_session_appointment apsa on apsa.action_plan_session_id = aps.id
+             inner join appointment app on app.id = apsa.appointment_id
+    where app.id = apo.id
+);
+
+Update appointment apo
+set referral_id = (
+    select ref.id
+    from referral ref
+             inner join supplier_assessment sup on sup.referral_id = ref.id
+             inner join supplier_assessment_appointment saa on saa.supplier_assessment_id = sup.id
+             inner join appointment app on app.id = saa.appointment_id
+    where app.id = apo.id
+)
+where apo.referral_id is null;
+
+alter table appointment
+    alter column referral_id set not null
```

Comment:
Will this cause problems during deployment? When the migration scripts are done and the new code yet to be deployed, any existing transactions will fail as they wont populate this field. Is this the case @tomsmyers?

Reply:
yeh i agree. i think we need to perform this step in a seperate deploy.

Reply:
i'll remove it

Reply:
Not sure that completely solves it. Because of the race condition, not all "referral ids" will get populated and so some will be null. Not sure how to solve this one